metis-core/metis-core-service/src/main/java/eu/europeana/metis/core/dao/DepublishedRecordDao.java

```diff
@@ -108,20 +111,20 @@ public int createRecordsToBeDepublished(String datasetId, Set<String> candidateR
     final long existingCount = countDepublishedRecordsForDataset(datasetId);
     if (existingCount + recordsToAdd.size() > maximumRecordsPerDataset) {
       throw new BadContentException(
-              "Can't add these records: this would violate the maximum number of records per dataset.");
+          "Can't add these records: this would violate the maximum number of records per dataset.");
     }
 
     // Add the records
     final List<DepublishedRecord> objectsToAdd = recordsToAdd.stream().map(recordId -> {
       final DepublishedRecord record = new DepublishedRecord();
       record.setDatasetId(datasetId);
       record.setRecordId(recordId);
-      record.setDepublicationState(DepublicationState.NOT_DEPUBLISHED);
+      record.setDepublicationStatus(DepublicationStatus.PENDING_DEPUBLICATION);
       return record;
     }).collect(Collectors.toList());
     ExternalRequestUtil.retryableExternalRequestConnectionReset(() -> {
       morphiaDatastoreProvider.getDatastore().save(objectsToAdd);
-      return null;
+      return Optional.empty();
```

Comment:
Either is perfectly correct since the return value is not used.
I changed it to start following the concept of not returning null values anymore, where possible..

Reply:
Ah, I see. SonarQube...

Reply:
Wasn't actually SonarQube, rather a common issue that I feel I've been facing for long with null returns, so thought perhaps to try this and see how it pans out with always returning optionals in cause of a return value being absent